mlflow/store/artifact/gcs_artifact_repo.py

```diff
@@ -86,6 +86,8 @@ def list_artifacts(self, path=None):
 
         results = bkt.list_blobs(prefix=prefix, delimiter="/")
         for result in results:
+            if result.name == prefix:
```

Comment:
@santosh1994 Can you insert a comment here to explain why we need this? May you could use https://github.com/googleapis/google-cloud-python/issues/5163 as a reference.

Reply:
done